enginetest/join_planning_tests.go

```diff
@@ -35,7 +36,7 @@ type JoinPlanTest struct {
 	indexes       []string
 	mergeCompares []string
 	exp           []sql.Row
-	order         []string
+	order         [][]string
```

Comment:
Add a docstring clarifying that the actual order can match any of the supplied orders?